GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -229,38 +229,83 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
                 }
             }
 
-            DiffHelper blobEnumerator = new DiffHelper(this.Tracer, this.Enlistment, this.FileList, this.FolderList, includeSymLinks: false);
+            BlockingCollection<string> availableBlobs = new BlockingCollection<string>();
+
+            ////
+            // First create the pipeline
+            //
+            //  diff ---> blobFinder ---> downloader ---> packIndexer
+            //    |           |              |                 |
+            //     ------------------------------------------------------> fileHydrator
+            ////
+            
+            // diff
+            //  Inputs:
+            //      * files/folders
+            //      * commit id
+            //  Outputs:
+            //      * RequiredBlobs (property): Blob ids required to satisfy desired paths
+            //      * FileAddOperations (property): Repo-relative paths corresponding to those blob ids
+            DiffHelper diff = new DiffHelper(this.Tracer, this.Enlistment, this.FileList, this.FolderList, includeSymLinks: false);
+
+            // blobFinder 
+            //  Inputs:
+            //      * requiredBlobs (in param): Blob ids from output of `diff`
+            //  Outputs:
+            //      * availableBlobs (out param): Locally available blob ids (shared between `blobFinder`, `downloader`, and `packIndexer`, all add blob ids to the list as they are locally available)
+            //      * MissingBlobs (property): Blob ids that are missing and need to be downloaded
+            //      * AvailableBlobs (property): Same as availableBlobs
+            FindMissingBlobsStage blobFinder = new FindMissingBlobsStage(this.SearchThreadCount, diff.RequiredBlobs, availableBlobs, this.Tracer, this.Enlistment);

```

Comment:
Rename to missingBlobsFinder?

Reply:
It's actually finding missing _and_ available blobs, so the class is the one that's misnamed. I'll fix it up.